### PR TITLE
[v1.x] License updates per ASF feedback

### PR DESCRIPTION
**Pull request by josephevans**
Address licensing concerns raised by our IPMC mentors:

1. Do not distribute 3rdparty/googletest/ci in source archive
2. Do not distribute docs/*** in source archive
3. Remove references to docs in LICENSE that are no longer distributed in source archive
4. Clarify include/mkldnn in LICENSE
5. Remove SIL open font license from licenses/ and reference in LICENSE

### CODE REVIEW

File: tools/source-exclude-artifacts.txt [hunk 1]
```diff
@@ -22,3 +22,5 @@
 
 R-package
 3rdparty/mkldnn/doc
+3rdparty/googletest/ci
+docs
```
Comment (leezu) [inline tools/source-exclude-artifacts.txt:26]: docs folder also contains the install instructions (build_from_source page)

**Review by samskalicky — APPROVED** [review]
thanks @josephevans

File: tools/source-exclude-artifacts.txt (hunk 1 above)
Comment (josephevans) [reply tools/source-exclude-artifacts.txt:26]: I would think most users building from source are following instructions from a web site, not viewing them from the source tree (after all, they are in .md format and would require viewing through a formatter/browser.) Do you think it's a requirement to ship them out in the source archive though?

**Review by ChaiBapchya — APPROVED** [review]
Thanks!

File: tools/source-exclude-artifacts.txt (hunk 1 above)
Comment (leezu) [reply tools/source-exclude-artifacts.txt:26]: I don't think it's a requirement, but it's not ideal as the website may change over time, while the archive remains fixed.

**Comment by ptrendx** [comment]
Removing entire docs directory just because of few fonts seems like quite a heavy handed approach. Could we instead just download those fonts as part of docs CD pipeline?

**Comment by josephevans** [comment]
> Removing entire docs directory just because of few fonts seems like quite a heavy handed approach. Could we instead just download those fonts as part of docs CD pipeline?

I agree that it's a fairly drastic change, but I am not convinced many users actually use the docs from the source archive. It takes a bit of effort to get them built and requires viewing them through a browser, and I would bet most people don't go through this effort and just use the documentation online.

File: tools/source-exclude-artifacts.txt (hunk 1 above)
Comment (josephevans) [reply tools/source-exclude-artifacts.txt:26]: True, but the git version tag will always be there.

**Review by Kh4L — APPROVED** [review]

File: tools/source-exclude-artifacts.txt (hunk 1 above)
Comment (kezhenxu94) [reply tools/source-exclude-artifacts.txt:26]: I think at least a link to the install instructions is needed

Comment (TristonC) [reply tools/source-exclude-artifacts.txt:26]: @josephevans Does doc cause the license issue?

Comment (TristonC) [reply tools/source-exclude-artifacts.txt:26]: Have you or have you heard anyone use the docs inside the release package?

Comment (TristonC) [reply tools/source-exclude-artifacts.txt:26]: @josephevans  Can we only keep the install instructions?

Comment (josephevans) [reply tools/source-exclude-artifacts.txt:26]: If you mean the build instructions, they are in https://github.com/apache/incubator-mxnet/blob/v1.x/docs/static_site/src/pages/get_started/build_from_source.md - we could copy this file to the root source archive. Alternatively, we could create a README.txt file with a link to the instructions on the web-site (but the latest version would have to be published.)

**Pull request merged**